### PR TITLE
convenient-test-manager: init at 1.2.1

### PR DESCRIPTION
**Pull request by algorithmiker**
###### Description of changes
add convenient-test-manager
https://pub.dev/packages/convenient_test
https://github.com/fzyzcjy/flutter_convenient_test

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -5138,6 +5138,12 @@
     githubId = 15957973;
     name = "Jeffry Molanus";
   };
+  gilice = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:5141]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

File: pkgs/development/tools/convenient_test_manager/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  src = fetchFromGitHub {
+    owner = "fzyzcjy";
+    repo = "flutter_convenient_test";
+    rev = "v${version}";
+    sha256 = "sha256-f0+nATpBAd+3ELRq7dfKi9HCawgF3LAAs0UqYjnKCPw=";
+  };
+  vendorHash = "sha256-0VZizwFlUfvrXLMn5iya2UMLcHhn5GtKaj2zekhqPS0=";
+  sourceRoot = "source/packages/${packageName}";
+  pname = packageName;
+  nativeBuildInputs = [ libvlc ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/convenient_test_manager/default.nix:24]:
```suggestion
flutter.mkFlutterApp {
  pname = "convenient-test-manager";
  version = "1.2.1";

  src = fetchFromGitHub {
    owner = "fzyzcjy";
    repo = "flutter_convenient_test";
    rev = "v${version}";
    sha256 = "sha256-f0+nATpBAd+3ELRq7dfKi9HCawgF3LAAs0UqYjnKCPw=";
  };

  vendorHash = "sha256-0VZizwFlUfvrXLMn5iya2UMLcHhn5GtKaj2zekhqPS0=";
  sourceRoot = "source/packages/convenient_test_manager";

  buildInputs = [ libvlc ];
```

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (gilice) [reply maintainers/maintainer-list.nix:5141]: thanks. Irrelevant now, because an other PR of mine was merged & added me as a maintainer

**Comment by gilice** [comment]
@SuperSandro2000 updated with your suggestions now.

File: pkgs/development/tools/convenient_test_manager/default.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    sha256 = "sha256-f0+nATpBAd+3ELRq7dfKi9HCawgF3LAAs0UqYjnKCPw=";
+  };
+
+  vendorHash = "sha256-0VZizwFlUfvrXLMn5iya2UMLcHhn5GtKaj2zekhqPS0=";
+  sourceRoot = "source/packages/convenient_test_manager";
+  postBuild = ''
+    mkdir -p $out/share/applications
+    cp $desktopItem/share/applications/*.desktop $out/share/applications
+    substituteInPlace $out/share/applications/*.desktop --subst-var out'';
+  buildInputs = [ libvlc ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/convenient_test_manager/default.nix:35]:
```suggestion
  src = fetchFromGitHub {
    owner = "fzyzcjy";
    repo = "flutter_convenient_test";
    rev = "v${version}";
    sha256 = "sha256-f0+nATpBAd+3ELRq7dfKi9HCawgF3LAAs0UqYjnKCPw=";
  };

  desktopItem = makeDesktopItem {
    name = pname;
    exec = "${placeholder "out"}/bin/convenient_test_manager";
    icon = "convenient_test_manager";
    desktopName = "Convenient Test Manager";
    genericName = "Flutter test runner";
    categories = [ "Development" ];
  };

  vendorHash = "sha256-0VZizwFlUfvrXLMn5iya2UMLcHhn5GtKaj2zekhqPS0=";

  nativeBuildInputs = [ copyDesktopItems ];

  buildInputs = [ libvlc ];

  sourceRoot = "source/packages/convenient_test_manager";
```

**Comment by gilice** [comment]
@SuperSandro2000 done, once again.

File: pkgs/development/tools/convenient_test_manager/default.nix [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  desktopItem = makeDesktopItem {
+    name = pname;
+    exec = "${placeholder "out"}/bin/convenient_test_manager";
+    icon = "convenient_test_manager";
+    desktopName = "Convenient Test Manager";
+    genericName = "Flutter test runner";
+    categories = [ "Development" ];
+  };
+
+  vendorHash = "sha256-0VZizwFlUfvrXLMn5iya2UMLcHhn5GtKaj2zekhqPS0=";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/convenient_test_manager/default.nix:30]:
```suggestion
  vendorHash = "sha256-iR/U2vQQqj95j1XkgW3/u7DQzwQrNmho9M4yLvefb4A=";
```

File: pkgs/development/tools/convenient_test_manager/default.nix [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  desktopItem = makeDesktopItem {
+    name = pname;
+    exec = "${placeholder "out"}/bin/convenient_test_manager";
+    icon = "convenient_test_manager";
+    desktopName = "Convenient Test Manager";
+    genericName = "Flutter test runner";
+    categories = [ "Development" ];
+  };
+
+  vendorHash = "sha256-iR/U2vQQqj95j1XkgW3/u7DQzwQrNmho9M4yLvefb4A=";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/convenient_test_manager/default.nix:30]:
```suggestion
  vendorHash = "sha256-GImFKbhAUOZn2R7274HqHaS3gq4bt/rhCUL+B+q2uXY=";
```

that hash is unstable. Can you check why that is the case?

Comment (SuperSandro2000) [reply pkgs/development/tools/convenient_test_manager/default.nix:30]: drafting it because of this
